spanner/api/Spanner.Samples/UpdateDataWithJsonAsync.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_update_data_with_json_column]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+public class UpdateDataWithJsonAsyncSample
+{
+    public class Venue
+    {
+        public int VenueId { get; set; }
+        public string VenueDetails { get; set; }
+    }
+
+    public async Task UpdateDataWithJsonAsync(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+        List<Venue> venues = new List<Venue>
+        {
+            new Venue { VenueId = 19, VenueDetails = @"{
+                ""rating"": ""9"",
+                ""description"": ""This is a nice place.""
+            }" },
+            new Venue { VenueId = 4, VenueDetails = @"[
+                {
+                    ""wing1"": { 
+                        ""description"": ""the first wing"",
+                        ""size"": ""5""
+                    }
+                },
+                {
+                    ""wing2"": {
+                        ""description"": ""the second wing"",
+                        ""size"": ""10""
+                    }
+                },
+                {
+                    ""main hall"": {
+                        ""description"": ""this is the biggest space"",
+                        ""size"": ""200""
+                    }
+                }
+            ]" },
+            new Venue { VenueId = 42, VenueDetails = @"{
```

Comment:
(AFAICT this version of) .NET does not have standard support for serializing an object to JSON, which means that this JSON literal needs to be written manually as a string. Another option could be to add a dependency on an external library that supports serializing objects to JSON, such as `NewtonSoft.JSON` [(example)](https://dzone.com/articles/quickly-create-json-object). That would simplify this sample, but would at the same time also introduce an external dependency.

@amanda-tarafa @jskeet 
What are your thoughts regarding this?

Reply:
I think it's absolutely fine to add a dependency on Json.NET for a *sample* - we can change to a different dependency later on without any of the normal versioning concerns, etc. I'd be much more nervous about taking a dependency in the production code.

Reply:
Google.Cloud.Spanner.Data already has a (transitive) dependency on Newtosoft.Json so I think it's totally fine to use that here. Also, in .NET, Newtonsoft.Json is a failry standard way to deal with Json.
Users in .NET Core 3.1 or newer should know whether they want to use System.Text.Json instead, so that's fine (Maybe leave a comment "If you are using .NET Core 3.1 ...")

Reply:
Great, thanks. I've updated the samples to use `NewtonSoft` for serialization, and added a comment about using `System.Text.Json` for newer versions of .NET.